reaper-adm-extension/src/reaper_adm/communicators.cpp

```diff
@@ -11,14 +11,18 @@ CommunicatorBase::~CommunicatorBase()
 }
 
 void CommunicatorBase::startSocket() {
-    auto resOpen = commandSocket.open();
-    assert(resOpen == 0);
-    auto resDial = commandSocket.dial(commandPort);
-    assert(resDial == 0);
-    resOpen = samplesSocket.open();
-    assert(resOpen == 0);
-    resDial = samplesSocket.dial(samplesPort);
-    assert(resDial == 0);
+    if(commandPort > 0) {
+        auto resOpen = commandSocket.open();
+        assert(resOpen == 0);
+        auto resDial = commandSocket.dial(commandPort);
+        assert(resDial == 0);
+    }
+    if(samplesPort > 0) {
+        auto resOpen = samplesSocket.open();
+        assert(resOpen == 0);
+        auto resDial = samplesSocket.dial(samplesPort);
+        assert(resDial == 0);
+    }
```

Comment:
It's related (needed to get the tests working) but also logic that should be in there anyway to prevent a crash if the Scene doesn't get instantiated for whatever reason. It's from commit a750833be8742557e61901ebc615266c5d24a5f3

Basically port 0 is the initial state. We only assign ports from 1 to 65535. The port numbers are changed when the extension successfully finds the Scene plugin and reads which ports it's using. So, if you attempt to start the communicator without first setting the port numbers (i.e, haven't been able to read them from the Scene plugin), it will not attempt to make a connection.